Issue #1532 Add support for Minikube ISO into integration tests

agajdosi opened the pull request:
Fix #1532

budhrg commented on test/integration/integration_test.go:
```diff
@@ -84,15 +84,15 @@ func TestMain(m *testing.M) {
 		isoUrl := os.Getenv("MINISHIFT_ISO_URL")
 		switch isoUrl {
-		case "b2d":
+		case "", "b2d":
```
:+1:

budhrg replied:
`00:06:56.424 Test run using Boot2Docker iso image.`

Test are running with b2d iso at the moment. Could you update the code to somehow pass minikube iso ? or `minikube` in case of empty `isoUrl` ?

agajdosi replied:
@budhrg So, are we switching for minikube.iso as default? If so I can then switch it and we will have no work with setting up env variable in centos.sh. Let me know.

budhrg commented on test/integration/integration_test.go:
```diff
@@ -794,14 +797,15 @@ func proxyLogShouldContainContent(expected *gherkin.DocString) error {
 func catDockerConfigFile() error {
 	var err error
-	if isoName == "b2d" {
+	switch isoName {
```
:+1:  for switch

agajdosi commented:
@budhrg Changed Minikube to be the default iso used with integration test as requested.

budhrg commented:
@agajdosi , few failures.

```
00:24:04.259   Scenario: Checking that docker-env value was applied  # features/cmd-config.feature:227
00:24:04.260     When printing Docker daemon configuration to stdout # integration_test.go:798 -> catDockerConfigFile
00:24:04.264     Command 'ssh -- cat /var/lib/systemd/system/docker.service' did not execute successfully. cmdExit: 1, cmdErr: cat: can't open '/var/lib/systemd/system/docker.service': No such file or directory
00:24:04.265 Command failed: exit status 1
00:24:04.265 
00:24:04.265     Then stdout should contain "FOO=BAR"                # integration_test.go:650 -> commandReturnShouldContain
00:24:04.269     And stdout should contain "hello=hi"                # integration_test.go:650 -> commandReturnShouldContain
00:24:04.491 
00:24:04.491   Scenario: Checking that docker-opt value was applied  # features/cmd-config.feature:232
00:24:04.492     When printing Docker daemon configuration to stdout # integration_test.go:798 -> catDockerConfigFile
00:24:04.494     Command 'ssh -- cat /var/lib/systemd/system/docker.service' did not execute successfully. cmdExit: 1, cmdErr: cat: can't open '/var/lib/systemd/system/docker.service': No such file or directory
00:24:04.495 Command failed: exit status 1
```

agajdosi commented:
@budhrg Thanks for info. Hmm, strange, on my machine it passed fine, will look into that in the morning.

amitkrout commented:
> Changed Minikube to be the default iso used with integration test as requested.

@agajdosi @budhrg Are you going to make it the permanent changes i mean b2d is no more going to be the default iso ?

budhrg commented:
@amitkrout @agajdosi , minikube changes are testing purpose only for this PR.

amitkrout commented:
@agajdosi Just realized, Let see how coolstore feature is running on minikube in ci. It will be like a full integration test including coolstore.

praveenkumar commented on Makefile:
```diff
@@ -190,7 +190,7 @@ integration: $(MINISHIFT_BINARY)
 integration_all: $(MINISHIFT_BINARY)
 	mkdir -p $(INTEGRATION_TEST_DIR)
-	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) --run-before-feature="$(RUN_BEFORE_FEATURE)" $(GODOG_OPTS) --tags ~coolstore
+	go test -timeout $(TIMEOUT) $(REPOPATH)/test/integration --tags=integration -v -args --test-dir $(INTEGRATION_TEST_DIR) --binary $(MINISHIFT_BINARY) --run-before-feature="$(RUN_BEFORE_FEATURE)" $(GODOG_OPTS) --tags ""
```
@agajdosi Instead of `--tags ""` we can just remove that option right? or you just did because to test this PR?

agajdosi replied:
@praveenkumar Just to test the `coolstore` out with minikube.iso, but it got stuck as normally. Removing it now.

agajdosi commented:
@budhrg @amitkrout Coolstore got stuck as it does, removing it now. Except coolstore the tests before it went fine, so removing coolstore and waiting for pass.

amitkrout commented:
Ahh..it got stuck due the old coolstore.yaml file which refers to some unknown registry images. The updated yaml file is in the pr #1519. This can not be merged due to inconsistency behavior in ci atm.

agajdosi commented:
This coolstore test was just a test to see how it works as requested. Now it does not contain any enabling of coolstore and works as expected.

@budhram I think this is ready, the only thing oa that before merge it needs to have line 85 in integration_test.go removed, this is here just to force CI to use minikube.iso.

LalatenduMohanty commented:
@agajdosi I think the PR is good. However my original issue was to also create an CI job with Minikube ISO on ci.centos.org in parallel with current CI job that's runs with B2D ISO. So that we gain confidence that every test that passes with b2d ISO also passes with Minikube ISO.  May be we can have a separate issue for that and track it there.

hferentschik commented:
We also need to setup a job for this. Maybe a nightly job to run against Minikube on CentOS CI?

agajdosi commented:
@LalatenduMohanty @hferentschik Sure, my idea was to add support to integration tests and then let @budhrg to continue with CI, make it 2 PRs for 1 issue.

But I can take a look on CentralCI and do it by myself, was thinking for some time that I would like to get some understanding of CI setup in upstream. @budhrg Let me know how you see it, if that would be unrealistic, lets take the CI part. (I can learn about CI later.)

hferentschik commented:
> But I can take a look on CentralCI and do it by myself, was thinking for some time that I would like to get some understanding of CI setup in upstream

+1  I think this would be a good idea.

LalatenduMohanty commented:
I am fine with handling the actual CI task in a separate issue (in this sprint). I will fix the description in #1532 to match the PR, so that we do not wait long for this PR to merge.

agajdosi commented:
@LalatenduMohanty +1 I have created a follow-up issue #1568 which is focused on creating a job in CI, please update later name of #1532 to reflect the part which was taken in this PR.

I have rebased and removed the line which forced the tests to use Minikube iso as default, now it will use b2d by default as before. Thus this is merge-able if approved.

LalatenduMohanty commented:
@agajdosi Thanks.
